ckanext/datastore/logic/action.py

```diff
@@ -433,16 +433,17 @@ def datastore_search(context, data_dict):
     :param include_total: True to return total matching record count
                           (optional, default: true)
     :type include_total: bool
-    :param total_estimation_threshold: By default, the "total" returned is a
-            precise count of the query results, which can be computationally
-            expensive for large results (e.g. >100,000k rows). By setting this
-            value to non-zero, and the estimated total is above this value,
-            then datastore_search will skip the precise count - it just returns
-            the estimated total. The estimated total is generated by EXPLAIN,
-            which is triggered by Express Loader or DataPusher as part of
-            loading the data. NB Total estimation is not compatible when
-            specifying 'filters' or 'distinct' options. (optional, default: 0)
-    :type total_estimation_threshold: int
+    :param total_estimation_threshold: If "include_total" is True and
+        "total_estimation_threshold" is not None and the estimated total
+        (matching record count) is above the "total_estimation_threshold" then
+        this datastore_search will return an *estimate* of the total, rather
+        than a precise one. This is often good enough, and saves
+        computationally expensive row counting for larger results (e.g. >100000
+        rows). The estimated total comes from the PostgreSQL table statistics,
+        generated when Express Loader or DataPusher finishes a load, or by
+        autovacuum. NB Total estimation is currently not compatible when
+        specifying 'filters' or 'distinct' options. (optional, default: None)
```

Comment:
the "not compatible" wording makes it sound like the number will be wrong if you pass total_estimation_threshold with filters or distinct. Won't filters or distinct just prevent the estimation from happening (like if total_estimation_threshold was set to None?)